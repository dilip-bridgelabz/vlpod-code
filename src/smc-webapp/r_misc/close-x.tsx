/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { Icon } from "./icon";

const closex_style: React.CSSProperties = {
  float: "right",
  marginLeft: "5px",
};

export function CloseX({
  on_close,
  style,
}: {
  on_close: () => void;
  style?: React.CSSProperties;
}) {
  const onClick = (e) => {
    if (e != undefined) {
      e.preventDefault();
    }
    on_close();
  };

  return (
    <a href="" style={closex_style} onClick={onClick}>
      <Icon style={style} name="times" />
    </a>
  );
}
