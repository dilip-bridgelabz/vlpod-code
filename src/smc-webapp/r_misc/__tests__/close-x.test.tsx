/*
 *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
 *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
 */

import * as React from "react";
import { shallow } from "enzyme";
import { CloseX } from "../close-x";

describe("smoke test close-x", () => {
  test("it renders", () => {
    function nothing() {
      return undefined;
    }
    const rendered = shallow(<CloseX on_close={nothing} />);
    expect(rendered).toMatchSnapshot();
  });

  test("it calls close on click", () => {
    const close_mock = jest.fn();

    const rendered = shallow(<CloseX on_close={close_mock} />);
    rendered.simulate("click");
    expect(close_mock.mock.calls.length).toBe(1);
  });

  test("it accepts custom styling for the Icon", () => {
    const close_mock = jest.fn();
    const custom_style: React.CSSProperties = {
      marginTop: "3px",
      marginBottom: "12px",
    };
    const rendered = shallow(
      <CloseX on_close={close_mock} style={custom_style} />
    );
    expect(rendered.children().prop("style")).toBe(custom_style);

    // Not sure if this assertion is any good. Splitting hairs.
    // This one isn't too brittle anyhow.
    expect(rendered.children()).toMatchSnapshot();
  });
});
