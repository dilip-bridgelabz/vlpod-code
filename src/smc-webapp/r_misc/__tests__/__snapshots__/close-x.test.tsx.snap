// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test close-x it accepts custom styling for the Icon 1`] = `
<Icon
  name="times"
  style={
    Object {
      "marginBottom": "12px",
      "marginTop": "3px",
    }
  }
/>
`;

exports[`smoke test close-x it renders 1`] = `
<a
  href=""
  onClick={[Function]}
  style={
    Object {
      "float": "right",
      "marginLeft": "5px",
    }
  }
>
  <Icon
    name="times"
  />
</a>
`;
